services/src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -162,14 +173,18 @@ public void run() {
       try {
         TopicState topicState = topicState();
         if (topicState == null) {
+          if(_topicCreationEnabled) {
+            Utils.createMonitoringTopicIfNotExists(_zkConnect, _topic, _configuredReplicationFactor, _partitionsToBrokerRatio);
+          }
           return;
         }
-        if (topicState.someBrokerWithoutLeader() || topicState.someBrokerMissingPartition() || topicState.insufficientPartitions()) {
+        if (topicState.someBrokerWithoutLeader() || topicState.someBrokerMissingPartition() || topicState.insufficientPartitions()
+            || topicState.insufficientReplicationFactor()) {
 
           LOG.info(_serviceName + ": topic rebalance started.");
           LOG.debug(_serviceName + ": broker count " + topicState._allBrokers.size() + " partitionCount " + topicState._partitionInfo.size() + ".");
 
-          if (topicState.replicationFactor() == -1) {
+          if (topicState.currentReplicationFactor() == -1) {
```

Comment:
Here the `topicState.currentReplicationFactor` is called 3 times and `topicState.insufficientReplicationFactor` is called 1 time. This makes code unnecessarily messy. Plus this is beyond the scope of this patch which says "Move topic creation from produce service to Topic Management Service".

At first glance I am not really convinced that we should have this check, and whether this is the best way to do this check. For things that is not straightforward, they should probably be put in their own pull request with proper explanation of why they are needed. Maybe you can remove it from this patch?